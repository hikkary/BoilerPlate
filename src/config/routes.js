import React from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import Login from '../containers/login'
import HomePage from '../containers/homePage'
import HouseCharacters from '../containers/houseCharacters'
import Ranking from '../containers/ranking'
import Options from '../containers/options'

export default class Routes extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Login} />
          <Route path="/home" component={HomePage} />
          <Route path="/house/:houseName" component={HouseCharacters} />
          <Route path="/ranking" component={Ranking} />
          <Route path="/options" component={Options} />
        </Switch>
      </Router>
    )
  }
}
