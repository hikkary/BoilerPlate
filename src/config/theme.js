export default {
  color: {
    black: '#222',
    white: '#fff',
    loginBackground: '#f2debd',
    bookSeparator: '#580303',
    primaryColor: '#7f0909',
    secondaryColor: '#940B0B',
    tertiaryColor: '#CB4B4B',
    gray: '#979797'
  }
}
