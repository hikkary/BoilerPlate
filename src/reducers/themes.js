import { CHANGE_THEME } from '../actions/themes'
import { themeLight } from '../config/themes'

const initialState = {
  currentTheme: themeLight
}

export default (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_THEME:
      return {
        ...state,
        currentTheme: action.theme
      }

    default:
      return state
  }
}
