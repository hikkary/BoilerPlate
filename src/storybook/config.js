import React from 'react'
import { addDecorator, configure } from '@storybook/react'
import { ThemeProvider } from 'styled-components'
import centered from '@storybook/addon-centered'
import '../../node_modules/react-vis/dist/style.css'

import theme from '../config/theme'

const ThemeDecorator = storyFn => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)
addDecorator(ThemeDecorator)
addDecorator(centered)

function loadStories() {
  require('../components/input/story.js')
  require('../components/loginForm/story.js')
  require('../components/panel/story.js')
  require('../components/texts/story.js')
  require('../components/graphTest/story.js')
  // You can require as many stories as you need.
}

configure(loadStories, module)
