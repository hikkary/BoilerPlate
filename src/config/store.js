import { applyMiddleware, createStore } from 'redux'
import logger from 'redux-logger'
// used for async func use on redux
// import thunk from 'redux-thunk'

import reducers from '../reducers'

const store = createStore(reducers, applyMiddleware(logger))

export default store
