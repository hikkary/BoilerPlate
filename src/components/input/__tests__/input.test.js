import React from 'react'
import Input from '../'
import renderer from 'react-test-renderer'
import { themeLight } from '../../../config/themes'
import { ThemeProvider } from 'styled-components'
import 'jest-styled-components'

class ThemeWrapper extends React.Component {
  render() {
    return (
      <ThemeProvider theme={themeLight}>{this.props.children}</ThemeProvider>
    )
  }
}

const component = renderer.create(
  <ThemeWrapper>
    <Input />
  </ThemeWrapper>
)
let tree = component.toJSON()

it('should match the snapshot', () => {
  expect(tree).toMatchSnapshot()
})

it('should be of type input', () => {
  console.log(tree)
  expect(tree.type).toEqual('input')
})
