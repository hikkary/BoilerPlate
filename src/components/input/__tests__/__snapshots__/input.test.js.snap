// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
.c0 {
  outline: none;
  border: none;
  border-bottom: 2px solid #7f0909;
  margin: 0px 0px 15px 0px;
  padding: 10px;
  font-size: 20px;
}

<input
  className="c0"
/>
`;
